.utils/generate_addon_structure.sh

```diff
@@ -0,0 +1,186 @@
+    usage
+    echo ""
+    echo "Error: Missing -n or -w" >&2
+    exit 1
+fi
+
+function print_green(){
+    GREEN='\033[0;32m'
+    echo -e "$GREEN$1"
+}
+
+function check_addon(){
+    if [ -d "../modules/$ADDON_NAME" ]; then
+        echo "Error: $ADDON_NAME folder already exists!"
+        exit 1
+    fi
+}
+
+function request_approve(){
+    echo ""
+    echo "##############################################################################"
+    echo "Please,"
+    echo "be advised, that this script uses relative paths and adds content to existing"
+    echo "files." 
+    echo ""
+    echo "You may not move this script to a different directory to ensure the proper"
+    echo "functioning."
+    echo "##############################################################################"
+    echo ""
+    read -r -p "Do you want to continue? [y/N] " response
+    case "$response" in
+        [yY][eE][sS]|[yY])
+            ;;
+        *)
+            exit 0
+            ;;
+    esac
+}
+
+function clone_template() {
+    git clone "https://github.com/equinix-labs/terraform-equinix-template.git" $ADDON_NAME
+}
+
+function override_template(){
+    ## remove not required files
+    files_rm=(".github/" ".gitignore" ".terraform.lock.hcl" "CODEOWNERS" "LICENSE")
+    for f in ${files_rm[@]}; do
+        rm -rf ./$ADDON_NAME/$f
+    done
+    ## add templates header
+    find ./$ADDON_NAME -type f -name "*.tf" -exec perl -pi -e 'print "# TEMPLATE: This file was automatically generated with \`generate_addon_structure.sh\`\n# TEMPLATE: and should be modified as necessary\n" if $. == 1' {} \;
+    find ./$ADDON_NAME -type f -iname "*.md" -exec perl -pi -e 'print "<!-- TEMPLATE: This file was automatically generated with `generate_addon_structure.sh` and should be modified as necessary -->\n" if $. == 1' {} \;
+    ## add/replace addon_specific_templates files
+    rsync -avP ./addon_specific_templates/ ./$ADDON_NAME/
+    ## get latest equinix provider version
+    get_equinix_provider_latest_release
+    ## replace variables
+    grep -rl "{EQUINIX_PROVIDER_VERSION}" ./$ADDON_NAME/ | xargs sed -i "" "s/{EQUINIX_PROVIDER_VERSION}/${EQUINIX_PROVIDER_VERSION}/g"
+    grep -rl "{ADDON_NAME}" ./$ADDON_NAME/ | xargs sed -i "" "s/{ADDON_NAME}/${ADDON_NAME}/g"
+    grep -rl "{ADDON_NAME^}" ./$ADDON_NAME/ | xargs sed -i "" "s/{ADDON_NAME^}/${CAPITALIZED_ADDON_NAME}/g"
+    grep -rl "{ADDON_WEBSITE_URL}" ./$ADDON_NAME/ | xargs sed -i "" "s/{ADDON_WEBSITE_URL}/${ADDON_WEBSITE_URL}/g"
+    ## remove templates extension
+    find ./$ADDON_NAME -depth -name "*.tpl" -exec sh -c 'mv "$1" "${1%.*}"' _ {} \;
+}
+
+function move_template(){
+    mv ./$ADDON_NAME/ ../modules/$ADDON_NAME
+}
+
+function get_equinix_provider_latest_release() {
+    EQUINIX_PROVIDER_REPO="equinix/terraform-provider-equinix"
+    EQUINIX_PROVIDER_VERSION=$(curl --silent "https://api.github.com/repos/$EQUINIX_PROVIDER_REPO/releases/latest" | grep '"tag_name":' | sed -E 's/.*"([^"]+)".*/\1/')
```

Comment:
Chris noticed that this is generating an invalid version string because the release tag starts with a `v`: https://github.com/equinix-labs/terraform-equinix-kubernetes-addons/pull/2/commits/edfe9f2e5ed87566a9029e9cee410e01590a6c31

Reply:
Fixed, thanks! 

https://github.com/equinix-labs/terraform-equinix-kubernetes-addons/blob/8e7ce00af6d2b0379a4ed859c192b82fcc73a4e2/.utils/generate_addon_structure.sh#L107